# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 9ee7acab98a4
# Full Hash: 9ee7acab98a470b2860125ba3deb439a64dc83e7
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-10 21:47:35
# Regressor Bug: 1529884
# File Overlap Count: 1
# Description:
#   Bug 1529884 - part 6: Through subject principal at Document::ExecCommand() to constructor of EditorBase::AutoEditActionDataSetter r=smaug
#   
#   `Document::ExecCommand()` knows subject principal.  This patch makes it tell
#   `EditorCommand::DoCommand()` and `EditorCommand::DoCommandParam()`.  Then,
#   makes they tell each editor public methods which may cause dispatching
# ==============================================================================

diff -r 258c4da663ca -r 9ee7acab98a4 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Mon Jun 10 10:26:04 2019 +0000
+++ b/dom/base/Document.cpp	Mon Jun 10 10:27:07 2019 +0000
@@ -4430,7 +4430,8 @@
   // require additional parameter, we can use `DoCommand()`.
   if (adjustedValue.IsEmpty() || paramType == EditorCommandParamType::None) {
     MOZ_ASSERT(!(paramType & EditorCommandParamType::Bool));
-    rv = editorCommand->DoCommand(commandData.mCommand, *maybeHTMLEditor);
+    rv = editorCommand->DoCommand(commandData.mCommand, *maybeHTMLEditor,
+                                  &aSubjectPrincipal);
     return !rv.ErrorCodeIs(NS_SUCCESS_DOM_NO_OPERATION) && !rv.Failed();
   }
 
@@ -4442,7 +4443,7 @@
                adjustedValue.EqualsLiteral("false"));
     rv = editorCommand->DoCommandParam(
         commandData.mCommand, Some(adjustedValue.EqualsLiteral("true")),
-        *maybeHTMLEditor);
+        *maybeHTMLEditor, &aSubjectPrincipal);
     return !rv.ErrorCodeIs(NS_SUCCESS_DOM_NO_OPERATION) && !rv.Failed();
   }
 
@@ -4454,7 +4455,7 @@
   if (!!(paramType & EditorCommandParamType::String)) {
     MOZ_ASSERT(!adjustedValue.IsVoid());
     rv = editorCommand->DoCommandParam(commandData.mCommand, adjustedValue,
-                                       *maybeHTMLEditor);
+                                       *maybeHTMLEditor, &aSubjectPrincipal);
     return !rv.ErrorCodeIs(NS_SUCCESS_DOM_NO_OPERATION) && !rv.Failed();
   }
 
@@ -4464,7 +4465,7 @@
     NS_ConvertUTF16toUTF8 utf8Value(adjustedValue);
     MOZ_ASSERT(!utf8Value.IsVoid());
     rv = editorCommand->DoCommandParam(commandData.mCommand, utf8Value,
-                                       *maybeHTMLEditor);
+                                       *maybeHTMLEditor, &aSubjectPrincipal);
     return !rv.ErrorCodeIs(NS_SUCCESS_DOM_NO_OPERATION) && !rv.Failed();
   }
 